CalibFormats/SiStripObjects/interface/SiStripClusterizerConditions.h

```diff
@@ -14,18 +14,21 @@ class SiStripClusterizerConditions {
 
   struct Det {
     bool valid() const { return ind != invalidI; }
-    uint16_t rawNoise(const uint16_t strip) const { return SiStripNoises::getRawNoise(strip, noiseRange); }
-    float noise(const uint16_t strip) const { return SiStripNoises::getNoise(strip, noiseRange); }
+    uint16_t rawNoise(const uint16_t strip) const { return rawNoises[strip]; }
+    float noise(const uint16_t strip) const { return 0.1f * float(rawNoises[strip]); }
     float weight(const uint16_t strip) const { return m_weight[strip / 128]; }
-    bool bad(const uint16_t strip) const { return quality->IsStripBad(qualityRange, strip); }
+    bool bad(const uint16_t strip) const { return qualityBits[strip]; }
     bool allBadBetween(uint16_t L, const uint16_t& R) const {
       while (++L < R && bad(L)) {
       };
       return L == R;
     }
+    static constexpr uint16_t kMaxStrips = 768;
```

Comment:
could one think about sparing some empty entries by using vectors and resizing at construction time with the correct number of strips (4 APVs vs 6 APVs)? The size is known via the detid.

Reply:
@mmusich 
do you happen to know if the number of APVs is strictly connected to the already available variables in the `SiStripClusterizerConditions::emplace_back` method: `invGains` or `connections` ?

Reply:
I think `invGains.size()` should know about it:

https://github.com/cms-sw/cmssw/blob/f0dc95b25faa4caa9ec87bdf0744500f8cb05ffe/RecoLocalTracker/SiStripClusterizer/plugins/SiStripClusterizerConditionsESProducer.cc#L68-L71

Reply:
> using vectors

I expect that access to `vector<bool>` is slower than `array<bool`. I guess I can use `vector<uint8_t>` or `char` to have a matching performance